app/src/main/java/app/revanced/integrations/patches/AlternativeThumbnailsPatch.java

```diff
@@ -18,25 +25,287 @@
+        LogHelper.printDebug(() -> "Using DeArrow API address: " + deArrowApiUrlPrefix);
+    }
+
+    /**
+     * Fix any bad imported data.
+     */
+    private static Uri validateSettings() {
+        final int altThumbnailType = SettingsEnum.ALT_THUMBNAIL_STILLS_TIME.getInt();
+        if (altThumbnailType < 1 || altThumbnailType > 3) {
+            ReVancedUtils.showToastLong("Invalid Alternative still thumbnail type: "
+                    + altThumbnailType + ". Using default");
+            SettingsEnum.ALT_THUMBNAIL_STILLS_TIME.resetToDefault();
+        }
+
+        Uri apiUri = Uri.parse(SettingsEnum.ALT_THUMBNAIL_DEARROW_API_URL.getString());
+        // Cannot use unsecured 'http', otherwise the connections fail to start and no callbacks hooks are made.
+        String scheme = apiUri.getScheme();
+        if (scheme == null || scheme.equals("http") || apiUri.getHost() == null) {
+            ReVancedUtils.showToastLong("Invalid DeArrow API URL. Using default");
+            SettingsEnum.ALT_THUMBNAIL_DEARROW_API_URL.resetToDefault();
+            return validateSettings();
+        }
+        return apiUri;
+    }
+
+    private static boolean usingDeArrow() {
+        return SettingsEnum.ALT_THUMBNAIL_DEARROW.getBoolean();
+    }
+
+    private static boolean usingVideoStills() {
+        return SettingsEnum.ALT_THUMBNAIL_STILLS.getBoolean();
+    }
+
+    /**
+     * Build the alternative thumbnail url using YouTube provided still video captures.
+     *
+     * @param decodedUrl Decoded original thumbnail request url.
+     * @return The alternative thumbnail url, or the original url. Both without tracking parameters.
+     */
+    @NonNull
+    private static String buildYoutubeVideoStillURL(@NonNull DecodedThumbnailUrl decodedUrl,
+                                                    @NonNull ThumbnailQuality qualityToUse) {
+        String sanitizedReplacement = decodedUrl.createStillsUrl(qualityToUse, false);
+        if (VerifiedQualities.verifyAltThumbnailExist(decodedUrl.videoId, qualityToUse, sanitizedReplacement)) {
+            return sanitizedReplacement;
+        }
+        return decodedUrl.sanitizedUrl;
+    }
+
+    /**
+     * Build the alternative thumbnail url using DeArrow thumbnail cache.
+     *
+     * @param videoId ID of the video to get a thumbnail of.  Can be any video (regular or Short).
+     * @param fallbackUrl URL to fall back to in case.
+     * @return The alternative thumbnail url, without tracking parameters.
+     */
+    @NonNull
+    private static String buildDeArrowThumbnailURL(String videoId, String fallbackUrl) {
+        // Build thumbnail request url.
+        // See https://github.com/ajayyy/DeArrowThumbnailCache/blob/29eb4359ebdf823626c79d944a901492d760bbbc/app.py#L29.
+        return dearrowApiUri
+                .buildUpon()
+                .appendQueryParameter("videoID", videoId)
+                .appendQueryParameter("officialTime", "true")
```

Comment:
It should not be included. It will do nothing in this case though since no timestamp is specified. It is used by the extension when requesting a thumbnail at a specific timestamp so that it knows to send back that timestamp by default in the future.